node/src/p2p/callbacks/p2p_callbacks_actions.rs

```diff
@@ -0,0 +1,56 @@
+use openmina_core::ActionEvent;
+use p2p::{
+    channels::{
+        rpc::{P2pRpcId, P2pRpcRequest, P2pRpcResponse},
+        streaming_rpc::P2pStreamingRpcResponseFull,
+    },
+    PeerId,
+};
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Debug, Clone, ActionEvent)]
+#[action_event(level = debug)]
+pub enum P2pCallbacksAction {
+    P2pChannelsRpcReady {
+        peer_id: PeerId,
+    },
+    P2pChannelsRpcTimeout {
+        peer_id: PeerId,
+        id: P2pRpcId,
+    },
+    P2pChannelsRpcResponseReceived {
+        peer_id: PeerId,
+        id: P2pRpcId,
+        response: Option<Box<P2pRpcResponse>>,
+    },
+    P2pChannelsRpcRequestReceived {
+        peer_id: PeerId,
+        id: P2pRpcId,
+        request: Box<P2pRpcRequest>,
+    },
+
+    P2pChannelsStreamingRpcReady,
+    P2pChannelsStreamingRpcTimeout {
+        peer_id: PeerId,
+        id: P2pRpcId,
+    },
+    P2pChannelsStreamingRpcResponseReceived {
+        peer_id: PeerId,
+        id: P2pRpcId,
+        response: Option<P2pStreamingRpcResponseFull>,
+    },
+}
+
+impl redux::EnablingCondition<crate::State> for P2pCallbacksAction {
+    fn is_enabled(&self, _state: &crate::State, _time: redux::Timestamp) -> bool {
+        match self {
+            P2pCallbacksAction::P2pChannelsRpcReady { .. } => true,
```

Comment:
I think these callbacks (that replace the interception of an action) are like effectful actions in the sense that they will always be enabled right? so the implementation here could be just `true`

Reply:
Yes, but I wrote it out just in case some of them needed it